src/utils/src/SerialDispatchQueue.c

```diff
@@ -0,0 +1,267 @@
+#include "Include_i.h"
+
+/**
+ * Create a serial dispatch queue object
+ */
+STATUS serialDispatchQueueCreate(PSERIAL_DISPATCH_QUEUE_HANDLE pHandle)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    PSerialDispatchQueue pSerialDispatchQueue = NULL;
+
+    CHK(pHandle != NULL, STATUS_NULL_ARG);
+
+    CHK_STATUS(serialDispatchQueueCreateInternal(&pSerialDispatchQueue));
+
+CleanUp:
+
+    if (STATUS_FAILED(retStatus)) {
+        pSerialDispatchQueue = NULL;
+    }
+
+    if (pHandle != NULL) {
+        *pHandle = TO_SERIAL_DISPATCH_QUEUE_HANDLE(pSerialDispatchQueue);
+    }
+
+    LEAVES();
+    return retStatus;
+}
+
+/**
+ * Frees and de-allocates the serial dispatch queue
+ */
+STATUS serialDispatchQueueFree(PSERIAL_DISPATCH_QUEUE_HANDLE pHandle)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    PSerialDispatchQueue pSerialDispatchQueue = NULL;
+
+    CHK(pHandle != NULL, STATUS_NULL_ARG);
+
+    pSerialDispatchQueue = FROM_SERIAL_DISPATCH_QUEUE_HANDLE(*pHandle);
+
+    serialDispatchQueueFreeInternal(&pSerialDispatchQueue);
+
+    // Set the handle pointer to invalid
+    *pHandle = INVALID_TIMER_QUEUE_HANDLE_VALUE;
+
+CleanUp:
+
+    LEAVES();
+    return retStatus;
+}
+
+STATUS serialDispatchQueueShutdown(SERIAL_DISPATCH_QUEUE_HANDLE handle)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    PSerialDispatchQueue pSerialDispatchQueue = NULL;
+
+    CHK(IS_VALID_SERIAL_DISPATCH_QUEUE_HANDLE_VALUE(handle), STATUS_INVALID_ARG);
+
+    pSerialDispatchQueue = FROM_SERIAL_DISPATCH_QUEUE_HANDLE(handle);
+
+    ATOMIC_STORE_BOOL(&pSerialDispatchQueue->shutdown, TRUE);
+    CVAR_SIGNAL(pSerialDispatchQueue->executorCvar);
+    THREAD_JOIN(pSerialDispatchQueue->executorTid, NULL);
+
+CleanUp:
+
+    LEAVES();
+    return retStatus;
+}
+
+STATUS serialDispatchQueueDispatchTask(SERIAL_DISPATCH_QUEUE_HANDLE handle,
+                                       DispatchQueueTaskFunc dispatchQueueTaskFn,
+                                       UINT64 customData,
+                                       DISPATCH_QUEUE_TASK_PRIORITY priority)
```

Comment:
To be consistent with other public APIs, make sure customData and dispatchQueuTaskFn in the right order. Also, move the priority as a second param - right after the handle.

Reply:
fixed